MapleServer2/PacketHandlers/Game/PremiumClubHandler.cs

```diff
@@ -64,12 +84,49 @@ private static void HandleOpenPurchaseWindow(GameSession session)
 
         private static void HandlePurchaseMembership(GameSession session, PacketReader packet)
         {
-            int packageId = packet.ReadInt();
-            session.Send(PremiumClubPacket.PurchaseMembership(packageId));
-            // TODO grab data from \table\vipgoodstable.xml for pricing, buff duration, and bonus items
-            long expiration = 2592847227; // temporarilyy plugging in expiration time
-            session.Send(PremiumClubPacket.ActivatePremium(session.FieldPlayer, expiration));
-            session.Player.IsVIP = true;
+            int vipId = packet.ReadInt();
+
+            if (!PremiumClubPackageMetadataStorage.IsValid(vipId))
+            {
+                return;
+            }
+
+            PremiumClubPackageMetadata vipPackage = PremiumClubPackageMetadataStorage.GetMetadata(vipId);
+
+            if (!session.Player.Wallet.Meret.Modify(vipPackage.Price))
+            {
+                return;
+            }
+
+            session.Send(PremiumClubPacket.PurchaseMembership(vipId));
+
+            foreach (BonusItem item in vipPackage.BonusItem)
+            {
+                Item bonusItem = new(item.Id)
+                {
+                    Rarity = item.Rarity,
+                    Amount = item.Amount
+                };
+                InventoryController.Add(session, bonusItem, true);
+            }
+
+            long expiration = DateTimeOffset.UtcNow.ToUnixTimeSeconds() + (vipPackage.VipPeriod * 3600); // Convert to seconds, as vipPeriod is given as hours
```

Comment:
You can make `vipPackage.VipPeriod * 3600` a variable then use here and on line 121